mesonbuild/backend/backends.py

```diff
@@ -1191,14 +1191,14 @@ def generate_depmf_install(self, d: InstallData) -> None:
     def get_regen_filelist(self) -> T.List[str]:
         '''List of all files whose alteration means that the build
         definition needs to be regenerated.'''
-        deps = [str(Path(self.build_to_src) / df)
-                for df in self.interpreter.get_build_def_files()]
+        deps = {str(Path(self.build_to_src) / df)
+                for df in self.interpreter.get_build_def_files()}
         if self.environment.is_cross_build():
-            deps.extend(self.environment.coredata.cross_files)
-        deps.extend(self.environment.coredata.config_files)
-        deps.append('meson-private/coredata.dat')
+            deps.update(self.environment.coredata.cross_files)
+        deps.update(self.environment.coredata.config_files)
+        deps.add('meson-private/coredata.dat')
         self.check_clock_skew(deps)
-        return deps
+        return list(deps)
```

Comment:
do we need to `sort()` here (or use an `OrderedSet`) to get reproducible ninja output?

Reply:
Do we generally need to have byte-identical `build.ninja` outputs? Do you mean something else?

Reply:
Yeah you're right, `OrderedSet` is good here. I somehow thought sets worked like dicts now and were iterated in insertion order.

Reply:
> Do we generally need to have byte-identical build.ninja outputs? Do you mean something else?

No, but if the same user regenerates build.ninja again with the same parameters and environment, the contents should be identical. Unordered iterables will break that and can cause an unnecessary regen / rebuild.

Reply:
Ah, makes sense. :)